app/src/utils/utils.ts

```diff
@@ -288,10 +299,13 @@ export const decodeMulticallReturnData = (
   calls.forEach((call) =>
     flatMap.push(
       ...call.fns.map((fn) => {
-        return call.contract.interface.decodeFunctionResult(
+        const data = call.contract.interface.decodeFunctionResult(
           typeof fn === 'string' ? fn : fn.fn,
           returnData[flatPtr++].returnData
-        )[0];
+        );
+        // If the call returned one element, just get rid of the array wrapper and return the data directly.
+        // If the call returned multiple elements (i.e. metaPtr returns an object), we want to preserve that structure
+        return data.length === 1 ? data[0] : data;
```

Comment:
Are we sure this a good idea ? while it make sense , now we have a function which returns an array or an object

Reply:
I have a few justifications for this:
1. If a call returns 1 element, there's no benefit to returning that single element as an array to the caller. They'll need to extract it from the array anyway, so we can do that here to reduce code/work by the caller. This removes a lot of useless code, as shown below
2. If a call returns multiple elements, you have to return all of them to the caller regardless so this makes sense
3. This approach minimizes the code diff / minimizes the risk of breaking things, since the change is contained within this method and the caller does not need to make any changes to how it parses return values

Details on 1:

Current format lets us do this: https://github.com/dcgtc/dgrants/blob/95a3bae36c62e7ac817d3537ab9f54adcb650fb0/app/src/utils/data/grantRounds.ts#L167-L177

If we always return the array, then we'd have to do this, which feels clunky for no benefit

```typescript
[
  _donationTokenAddress,
  _startTime,
  _endTime,
  _metadataAdmin,
  _payoutAdmin,
  _registryAddress,
  metaPtr,
  _hasPaidOut,
  _funds,
] = await callMulticallContract(...)

const donationTokenAddress = donationTokenAddress[0]
const startTime = startTime[0]
const endTime = endTime[0]
const metadataAdmin = metadataAdmin[0]
const payoutAdmin = payoutAdmin[0]
const registryAddress = registryAddress[0]
const hasPaidOut = hasPaidOut[0]
const funds = funds[0]
```